int64 id
bool valid

float64 x
float64 y
float64[4] noise_cov
float64[3] color
